aea/mail/base.py

```diff
@@ -395,7 +394,7 @@ def __str__(self):
         )
 
 
-class Multiplexer:
```

Comment:
Please do not relabel this class. This would break backwards compatibility and is also not desirable, the multiplexer should by definition be async so it does not need it in the class name.

Reply:
@DavidMinarsch 
it's done for compatibility on the first stage.
Multiplexer turned completely async and incompatible with all current code and tests
for this purpose it's renamed to AsyncMultiplexer, Multiplexeit is a sublaclass of async version with full compatible interface for current code.
after switching to runtime, only async version will stay and  named Multiplexer

Reply:
Thanks for explanation. I suggest we maintain the two classes. We want to be able to run the AEA in two modes: one single threaded mode were the event loop is shared between multiplexer and agent loop, another mode with two threads, one for the multiplexer, one for the agent loop. It seems to me the `Multiplexer` class can then in mode one directly instantiate the `AsyncMultiplexer` and in mode two implement its current functionality.

Reply:
@DavidMinarsch 
the difference is pretty small between to classes, just three interface functions, so need to find how to use those two interfaces and select it. Suppose it will be solved on making runtimes switchable